Makefile

```diff
@@ -1,7 +1,6 @@
-CC = gcc
-AR = ar
-override CFLAGS := -std=gnu99 -D_FILE_OFFSET_BITS=64 -Wall -Wextra -O3 -g -MMD $(CFLAGS)
-LDFLAGS =
+CC ?= gcc
+AR ?= ar
```

Comment:
My point is that `?=` is always noop, `CC` cannot be undefined so conditional assignment will never work. I was saying that `make` is clever enough, because `make CC=clang` will enforce this value even if `CC = gcc` assignment is present in the `Makefile`

``` shell
$ echo -e 'CC ?= gcc' > Makefile
$ echo -e 'all:' >> Makefile
$ echo -e '\t@echo $(CC)' >> Makefile
# Current makefile
$ make
cc
$ make CC=clang
clang
$ sed -i 's/\?=/=/' Makefile
$ make
gcc
$ make CC=clang
clang
```